harness/determined/common/api/_rbac.py

```diff
@@ -0,0 +1,135 @@
+from typing import Dict, List, Optional, Tuple
+
+from determined.common import api
+from determined.common.api import bindings
+
+
+def role_name_to_role_id(session: api.Session, role_name: str) -> int:
+    # XXX: is 499 guaranteed to be enough?  Why not read_paginated?
+    req = bindings.v1ListRolesRequest(limit=499, offset=0)
+    resp = bindings.post_ListRoles(session=session, body=req)
+    for r in resp.roles:
+        if r.name == role_name and r.roleId is not None:
+            return r.roleId
+    raise api.errors.BadRequestException(f"could not find role name {role_name}")
+
+
+def create_assignment_request(
+    session: api.Session,
+    role_name: str,
+    workspace_name: Optional[str] = None,
+    username_to_assign: Optional[str] = None,
+    group_name_to_assign: Optional[str] = None,
+) -> Tuple[List[bindings.v1UserRoleAssignment], List[bindings.v1GroupRoleAssignment]]:
+    if (username_to_assign is None) == (group_name_to_assign is None):
+        # XXX: very weird that we use api.errors.BadRequestException here!
```

Comment:
Don't forget to either remove me or turn this into a TODO and/or ticket.

Reply:
ah yes, thank you